security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/SubContingenciesProvider.java

```diff
@@ -41,7 +43,10 @@ public SubContingenciesProvider(ContingenciesProvider delegate, Partition partit
         int fullSize = fullList.size();
         int start = partition.startIndex(fullSize);
         int end = partition.endIndex(fullSize);
-        return new ArrayList<>(fullList.subList(start, end));
+        if (start == 0 && end == 0) {
+            return new ArrayList<>(Collections.emptyList());
+        } else {
+            return new ArrayList<>(fullList.subList(start, end));
+        }
```

Comment:
If `start` and `end` are equal to zero, is this code equivalent to:
```
return new ArrayList<>(fullList.subList(start, end));
```

If the answer is `yes`, revert this change.